testim-created/unzip-files-validation.js

```diff
@@ -0,0 +1,143 @@
+    throw new Error('Enter a file dirctory');
+}
+
+if (typeof encoding === '' || typeof encoding === 'undefined') {
+    throw new Error('Enter valid encoding string');
+}
+// extract zip file and load into an array of records (files)
+console.log('File name is ' +filename);
+let zip = new AdmZip(filename);
+zipEntries = zip.getEntries();   
+
+// zip entries is an object so to compare files there is only need for the file name 
+// get file name and load into new array
+zipEntries.forEach(file => newArray.push(file.name)); 
+
+// perform valiation loop over array of extracted files
+for (let i=0; i<= newArray.length; i++) {
+
+     // if the extracted file name matches with our expected file name then set the result to true and add  to the results array
+     // otherwise set the result to false
+     
+    if (newArray[i] === fileArray[i]) {
+        console.log(`File from extracted zip ${newArray[i]} is the same as file to validate: ${fileArray[i]}`);
+        results.push(true);
+    } else {
+         console.log(`File from extracted zip ${newArray[i]} is not the same as file to validate: ${fileArray[i]}`);
+         results.push(false);
+    }
+}
+
+// print out file content which could be used for validation of what is in a file as well expectedFileContent
+// if there is a need to loop all files uncomment 
+
+/*
+for (let i=0; i<= zipEntries.length; i++ ) {
+
+    let extractedFileContent = zipEntries[i].getData().toString(encoding);
+    console.log('Extracted file data: ' +extractedFileContent);
+
+    if (extractedFileContent === expectedFileContent[i]) {
+         results.push(true);
+    } else {
+         console.log(`Extracted data: ${extractedFileContent} did not match expected data: ${expectedFileContent[i]}`);
+         results.push(false);
+    }
+}
+*/
+
+zipEntries.forEach(file => console.log(`File Name ${file.name} File content ${file.getData().toString(encoding)}`));
+
+let myFirstFileContent = zipEntries[0].getData().toString(encoding);
+//console.log('First file content = ' +myFirstFileContent);
+
+let mySecondFileContent = zipEntries[1].getData().toString(encoding);
+//console.log('Second file content = ' +mySecondFileContent);
+
+if (myFirstFileContent = expectedFileContent[0]) {
+            results.push(true);
+} else {
+            console.log(`Extracted file contents: ${myFirstFileContent} does\'nt match expected file contents: ${expectedFileContent[0]}`);
+            results.push(false);
+}
+
+if (mySecondFileContent = expectedFileContent[1]) {
+           // console.log('First file content = ' +myFirstFileContent);
+            results.push(true);
+} else {
+            console.log(`Extracted file contents: ${mySecondFileContent} does\'nt match expected file contents: ${expectedFileContent[1]}`);
+            results.push(false);
+}
+
+
+
+// validate the results array and check if there are any false entries if so throw an error and fail the step 
+// in testim.io 
+
+console.log(results);
+
```

Comment:
This is just going to write out an array of true/false.  Would be nice to include file name